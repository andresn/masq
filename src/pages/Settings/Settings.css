.Settings {
  /* padding-bottom: 64px; */
  display: grid;
  grid-template-columns: auto 300px;
  width: 100%;
  height: 100%;
}

.Settings .title-container {
  margin-bottom: 32px;
}

.Settings .Avatar {
  width: 192px;
  height: 192px;
}

.Settings .profile {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
  align-items: start;
  grid-gap: 32px;
}

.Settings .inputs {
  display: flex;
  flex-direction: column;
  height: 192px;
  justify-content: space-between;
}

.Settings .fields {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.Settings .TextInput {
  width: 224px;
}

.Settings .button-container {
  display: flex;
  justify-content: center;
}

.Settings .parameters p {
  margin: 0;
  font-size: 14px;
  line-height: 1.29;
  letter-spacing: 0.7px;
  text-align: left;
  color: var(--grey3-color);
  max-width: 277px;
}

.Settings .sidebar {
  text-align: center;
}

.Settings .appSettings {
  margin-top: 32px;
}

.Settings .buttons {
  display: flex;
  align-items: center;
}

.Settings .exportButton {
  margin-right: 32px;
}
