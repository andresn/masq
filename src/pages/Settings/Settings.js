import React from 'react'
import PropTypes from 'prop-types'

import { Avatar, Button, Separator, TextInput, SwitchButton, CircleIndicator } from 'components'

import { UserContext } from 'context/user'

import './Settings.css'

class Settings extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      image: props.user.image,
      lastname: props.user.lastname,
      firstname: props.user.firstname,
      username: props.user.username
    }
    this.validate = this.validate.bind(this)
  }

  onChange (field, event) {
    this.setState({[field]: event.target.value})
  }

  onImageChange (event) {
    const reader = new window.FileReader()
    const file = event.target.files[0]
    reader.addEventListener('load', () => {
      this.setState({image: reader.result})
    })
    reader.readAsDataURL(file)
  }

  validate () {
    const isValid = !Object.values(this.state).some(val => !val)
    if (!isValid) return window.alert('Invalid form')
    this.props.onUpdateUser(this.state)
  }

  render () {
    const { onDeleteUser } = this.props

    return (
      <div className='Settings'>
        <h2 style={{marginLeft: '16px'}}>Your Profile</h2>

        <div className='profile'>
          <Avatar upload image={this.state.image} onChange={(e) => this.onImageChange(e)} />
          <div className='fields'>
            <TextInput label='Last Name' defaultValue={this.state.lastname} onChange={(e) => this.onChange('lastname', e)} />
            <TextInput label='First Name' defaultValue={this.state.firstname} onChange={(e) => this.onChange('firstname', e)} />
            <TextInput label='Username (Displayed)' defaultValue={this.state.username} onChange={(e) => this.onChange('username', e)} />
          </div>
        </div>

        <h2 style={{marginLeft: '16px'}}>Masq Parameters</h2>

        <div className='parameters'>
          <CircleIndicator style={{alignSelf: 'flex-start'}} color='#458bf8' />
          <p>Accept new applications automatically</p>
          <SwitchButton checked secondary color='#458bf8' />
        </div>

        <Separator height='48px' />
        <div className='button-container'>
          <Button label='SAVE CHANGES' onClick={this.validate} />
        </div>

        <Separator height='16px' />
        <div className='button-container'>
          <Button label='DELETE MY ACCOUNT' onClick={onDeleteUser} />
        </div>
      </div>
    )
  }
}

export default props => (
  <UserContext.Consumer>
    {user => <Settings {...props} user={user} />}
  </UserContext.Consumer>
)

Settings.propTypes = {
  user: PropTypes.object.isRequired
}
