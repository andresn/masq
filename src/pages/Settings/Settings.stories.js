import React from 'react'
import { UserContext } from 'contexts/user'
import { storiesOf } from '@storybook/react'
import { I18nextProvider } from 'react-i18next'

import i18n from 'i18n/i18n'

import Settings from './Settings'

const user = {
  image: 'https://randomuser.me/api/portraits/women/79.jpg',
  firstname: 'Alicia',
  lastname: 'Ford',
  username: 'Mustang'
}

storiesOf('Settings Page', module)
  .add('settings', () => (
    <I18nextProvider i18n={i18n}>
      <UserContext.Provider value={user}>
        <Settings user={user} />
      </UserContext.Provider>
    </I18nextProvider>
  ))
