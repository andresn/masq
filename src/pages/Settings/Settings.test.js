import React from 'react'
import renderer from 'react-test-renderer'

import { UserContext } from 'context/user'

import Settings from './Settings'

it('renders Settings page', () => {
  const user = {
    image: 'https://randomuser.me/api/portraits/women/79.jpg',
    firstname: 'Alicia',
    lastname: 'Ford',
    username: 'Mustang'
  }
  const component = renderer.create(
    <UserContext.Provider value={user}>
      <Settings user={user} />
    </UserContext.Provider>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
