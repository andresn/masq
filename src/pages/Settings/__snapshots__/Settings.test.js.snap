// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Settings page 1`] = `
<div
  className="Settings"
>
  <h2
    style={
      Object {
        "marginLeft": "16px",
      }
    }
  >
    Your Profile
  </h2>
  <div
    className="profile"
  >
    <div>
      <input
        accept=".jpg, .jpeg, .png"
        name="avatar"
        onChange={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
        type="file"
      />
      <div
        className="Avatar upload"
        onClick={[Function]}
        style={
          Object {
            "backgroundImage": "url(https://randomuser.me/api/portraits/women/79.jpg)",
          }
        }
      >
        <svg
          className="camera"
          color="white"
          height={33}
          viewBox="0 0 24 20"
          width={40}
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <defs>
            <path
              d="M21 3h-3.5L15.8.4c-.2-.2-.5-.4-.8-.4H9c-.3 0-.6.2-.8.4L6.5 3H3C1.3 3 0 4.3 0 6v11c0 1.7 1.3 3 3 3h18c1.7 0 3-1.3 3-3V6c0-1.7-1.3-3-3-3zm1 14c0 .6-.4 1-1 1H3c-.6 0-1-.4-1-1V6c0-.6.4-1 1-1h4c.3 0 .6-.2.8-.4L9.5 2h4.9l1.7 2.6c.3.2.6.4.9.4h4c.6 0 1 .4 1 1v11zM12 6c-2.8 0-5 2.2-5 5s2.2 5 5 5 5-2.2 5-5-2.2-5-5-5zm0 8c-1.7 0-3-1.3-3-3s1.3-3 3-3 3 1.3 3 3-1.3 3-3 3z"
              id="camera-svg"
            />
          </defs>
          <use
            fill="currentColor"
            fillRule="nonzero"
            xlinkHref="#camera-svg"
          />
        </svg>
      </div>
    </div>
    <div
      className="fields"
    >
      <div
        className="TextInput"
      >
        <input
          defaultValue="Ford"
          onChange={[Function]}
          type="text"
        />
        <label>
          Last Name
        </label>
      </div>
      <div
        className="TextInput"
      >
        <input
          defaultValue="Alicia"
          onChange={[Function]}
          type="text"
        />
        <label>
          First Name
        </label>
      </div>
      <div
        className="TextInput"
      >
        <input
          defaultValue="Mustang"
          onChange={[Function]}
          type="text"
        />
        <label>
          Username (Displayed)
        </label>
      </div>
    </div>
  </div>
  <h2
    style={
      Object {
        "marginLeft": "16px",
      }
    }
  >
    Masq Parameters
  </h2>
  <div
    className="parameters"
  >
    <div
      className="CircleIndicator"
      style={
        Object {
          "border": "solid 2px #458bf8",
        }
      }
    />
    <p>
      Accept new applications automatically
    </p>
    <div
      className="SwitchButton"
    >
      <div
        className="secondary"
        id="switch-container"
        onClick={[Function]}
      >
        <div
          className="secondary"
          id="background"
          style={
            Object {
              "backgroundColor": "#458bf8",
            }
          }
        />
        <div
          className="secondary"
          id="cursor"
          style={
            Object {
              "transform": "translate3d(32px, 0, 0)",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    id="separator"
    style={
      Object {
        "marginTop": "48px",
      }
    }
  />
  <div
    className="button-container"
  >
    <button
      className="Button primary"
      onClick={[Function]}
    >
      SAVE CHANGES
    </button>
  </div>
  <div
    id="separator"
    style={
      Object {
        "marginTop": "16px",
      }
    }
  />
  <div
    className="button-container"
  >
    <button
      className="Button primary"
      onClick={undefined}
    >
      DELETE MY ACCOUNT
    </button>
  </div>
</div>
`;
